### ISSUE TITLE
F-measure as the paper

### ISSUE THREAD

**Issue by kobybibas** [ben-hayes/beat-tracking-tcn#3]
Hi Ben,
I'm trying to evaluate the default model performance as in the paper's TABLE III.

I did the following.
1. Downloaded the Ballroom dataset and annotations.
2. Created the spectrograms using "./scripts/create_spectrograms.py" script.
3. Evaluate the default model on the entire Ballroom dataset using "./scripts/evaluate_model.py". 

I got the following metric
`F-msr    |  Cmgl  | CmglBstMtrcLvl     |   Gt      | P-scr `
`0.364 | 0.2447 | 0.347892                 | 0.0115 | 0.4151 `

However, in the paper's TABLE III the F-measure is 0.933.

Did you succeed to recreate the paper's performance?
Did I skip something?

And again, I really appreciate this repository and your time in developing it.
Koby

**Comment by ben-hayes** [comment]
Hey @kobybibas 

Thanks for the comment and the detailed info on the steps you took. It's a few months since I put this repo together now so my memory is a little hazy, but those scores definitely seem wrong.

It's worth pointing out, by the way, that this model is not an exact recreation of the model in the Davies & Böck paper, as it introduces residual connections and performs two convolutions at each temporal dilation step. As a result of these changes I was able to get the model to slightly outperform Davies & Böck's model on f-measure and Cemgil accuracy.

I'll have to look through my notes to be sure, but if I remember correctly I achieved those scores by averaging metrics across an 8-fold cross-validation (there should be scripts for this in the repo) — this is how evaluation was performed in the original paper. It might be a week or two until I'm able to get round to it, but I'll try to retrace my steps and see if I can recreate the problem you're having.

Cheers
Ben

**Comment by kobybibas** [comment]
Hi,
I didn't consider that following  #1  there is a need to retrain the default model. 
By using the code without the fix of #1, the performance is as reported at the paper.

Best,
Koby

**Comment by ben-hayes** [comment]
Aha! That will do it. Good catch.

Glad you've got it working.

Cheers
Ben

**Comment by tshpakova** [comment]
Hello, thank you for the repo, it's great.

I have similar issue, I try to reproduce the results, but I achieve around 0.3 F-measure with default checkpoint.
How can I get 0.9 F-measure? Should I train the model myself or there is a pretrained model?

Thanks in advance,
Tatiana

**Comment by iranroman** [comment]
Hello @kobybibas or @ben-hayes . Thank you very much for your efforts to bring this important model to pytorch. 

Would any of you have and be willing to share the model weights that replicate the F-measures quoted in the paper? I know that this model is a bit different and improved, so having the most optimal parameters would be awesome. 

Thank you!

**Comment by nikhilsos** [comment]
Hello @iranroman were you able to get F-measures quoted in the paper? As @kobybibas quoted above I though maybe retraining the model would fix the f-measure but I could get only 0.4 F measure on ballroom dataset without averaging metrics across an 8-fold cross-validation,  and using the stopping condition mentioned.  Since F measure I got was 0.4 I doubt 8 fold training will give the F measure quoted in the paper. If you managed to replicate the F measure would you mind telling me the training conditions? or am I doing something wrong? I know it's been 2 years, but I'd  greatly appreciate any response.

Thanks.

**Comment by nikhilsos** [comment]
> Hi, I didn't consider that following #1 there is a need to retrain the default model. By using the code without the fix of #1, the performance is as reported at the paper.
> 
> Best, Koby

Sorry? Without the fix of #1 ? Does that mean the f_measure is as mentioned in paper without the fix? without using 2 convolutional layers?
